tests/ui/issues/issue-42880.stderr

```diff
@@ -3,6 +3,13 @@ error[E0599]: no associated item named `String` found for struct `String` in the
    |
 LL |     let f = |&Value::String(_)| ();
    |                      ^^^^^^ associated item not found in `String`
+   |
+note: if you're trying to build a new `String` consider using one of the following associated functions:
```

Comment:
This note seems wrong.

Reply:
Earlier there's a `type Value = String;`, which means this is `String::String(_)`.

Reply:
That's not the problem here. This is a bug having to do with a bug in a *pattern*, but we're emitting associated *function* suggestions for it.

You can suppress this by checking if `args` is some:
https://github.com/rust-lang/rust/blob/fb61292105708ea45cf5a4ed596dbbd54c8ed429/compiler/rustc_hir_typeck/src/method/suggest.rs#L127

Reply:
`args.is_none()` in *all* cases that should trigger :-/

I'll see if I can find another way of identifying this case.

Reply:
Solved by looking at the parent hir node.